### ISSUE TITLE
Feature : Task assignment with pay tracking

### ISSUE THREAD

**Issue by dstevens79** [recursivetree/seat-alliance-industry#20]
lmeve is absolutely amazing, Lmeve is on github and written in php. it is out of date, and while still works.. kinda only barely so. I cannot find a dev to pick it up, but this is very close to that in functionality and if you were to just alter this a little bit with a few tiny features and this could replace lmeve in full...

In its current form this plugin is a free for all which is great for buddy buddy, and the ability to put up global tasks anyone could take was a request on lmeve, but lets talk about corporate level focused industrail efforts : 

whats missing?

A method to assign "build this / this fit" to members with 
          a management page for assigning jobs (without a stated payout value)
                    the option to build x and in y amounts
                    an option to require stated y volume of x to be built as a monthly assignment

A method to track work and completion rates total and per month for tasks assigned
          Lmeve uses a list sortable by individual members or the whole corp.
          Lmeve also lets members see thier own jobs only if they do not have management rights
          Lmeve also stores completions in a month to month reviewable way so you can go back and see what was accomplished
          
![image](https://user-images.githubusercontent.com/29002807/222610982-ea30dab4-ec02-47e0-8796-9a03c85083e0.png)
![image](https://user-images.githubusercontent.com/29002807/222611279-424ed675-a4e0-44e3-a6fa-b717ec2f237e.png)

An internal means to track work done that earns members income : 
          Lmeve uses an adjustable point per hour system for each industry task 
          then the combined points get converted to isk 
          
![image](https://user-images.githubusercontent.com/29002807/222610761-60c1abef-5dd4-4a9e-9178-02febcbda71f.png)

Lmeve never implemented the ability to filter members to assign a job to based on thier industrial skills, but that was in the works when the project started to stall out in development.

This is the closest thing i have seen to it, and there are alot of people that having this kind of functionality working properly again and readily available to the eve community would make the lives of so many corps and corp members so much better.

**Comment by recursivetree** [comment]
It's nice to hear that you find the plugin useful, and I appreciate the feedback on how to improve it. However, just to make it clear: This plugin is not lmeve and will not become lmeve. Still, I think it is possible to integrate most or at least some features into seat-alliance-industry.

As a response to your feature requests:
* An option to assign orders to certain members should be possible.
* Currently, prices are always dictated by the order creator. Changing that to be delivery based is not impossible, but it brings some issues, especially with the system of guards that ensures that prices are fair. Besides that, you can also just ignore the price and say it's not meaningful in your corporation/alliance/group. I get that that's not ideal, but it's better than nothing.
* Regarding monthly assignments, I guess adding a feature that creates a new order on a monthly basis is enough?
* Statistics are already requested here: https://github.com/recursivetree/seat-alliance-industry/issues/11 . maybe discuss it there?
* There won't be a point system. I just straight out want to say this.
* Assigning jobs based on skills is something that sounds interesting as an extension once assigning jobs to members is done

In general, you're asking for some major features. Expect some discussion around how it's implemented(this is not lmeve) and it might take a while to get everything done. If you can provide a priority list, I can try to get the most important things done first.

Also, please open individual issues for each feature request to track progress and discussion independent from each other.

**Comment by dstevens79** [comment]
I completely agree, this is not lmeve...
I need some time to think on all of the points 

but loosely 

The overall goal is to generally be able to assign industrial work to individuals and track that it gets completed.
If specific jobs can be assinged a monetary value by invested hours completed for an industry job type (copy, invention, etc) then points are not required, one of the unique perks of lmeve was the ability to create a motivator through income based on assigned work that is completed.

and this bit is probably the core of the whole enhancement, 
if the features of : "assign", "track" exist then its baseline useable as a corporate industrial management tool.
If the feature of : "isk math per invested job hour by type" exists then its a corporate industrial management tool with a nice perk that appeals to user members as well as admin members.

**Comment by recursivetree** [comment]
How do you imagine "isk math per invested job hour by type"? Getting the time for the final assembly is easy, but if you want it to consider time for it's ingredients(like for a capital) as well, you are asking for a fully fledged industry calculator like ravworks. And that won't happen. seat-alliance-industry is a marketplace, not an industry calculator. Building a good industry calculator is just out of scope.

**Comment by dstevens79** [comment]
Just take completed corp industry jobs and do a  time to completion * some isk value we can configure per job type.
its easy to tell how long a job takes, runtimes are part of the print data, or you can ping esi every 15 mins to see "is person running or completed an assigned corp job" Also - this would only track assigned jobs not all the jobs across the corp.


example : 
we assign a member a few jobs... he does some of them...

for that member :
total time completed of assigned manufacturing = 35 hrs * (custom isk rate set for manu)  == total isk value for manufacturing earned.
total time completed of assigned copy jobs = 350 hrs * (custom isk rate set for copying)  == total isk value for copying earned.

Then add the earned isk value totals as a final earned total for that month.
That total is what the pilot earned for the work they did for the corp for that month that an officer can pay out in eve.


Its definately not an industry calculator - its a directed activity tracker, tracking thier invested industry time of only what was assigned for them to do. i could probably do the basic math if you can just count and display the invested time per pilot per month, but if you are able to count and display that then i cant imagine it would be a complex problem applying a custom isk value multiplier to the hour totals for each job type?

**Comment by recursivetree** [comment]
Great, I'm sure I can add an option to value orders by industry time instead of market value. Rather than taking the job times from running jobs over the ESI, is it fine if I take it from the SDE? This is way simpler and less error prone. Additionally, what happens if a user didn't start any order for a job assigned to him? There wouldn't be a price when the order is created, yet the plugin requires it.

**Comment by dstevens79** [comment]
make a default price 0.01 isk possibly,
there is an esi route to know if a job was completed, so the check could be something like 
"if pilot has x manufacturing tasked, does corp industry show this as completed, if so how many were done"

**Comment by recursivetree** [comment]
I can add a default price, yes.
The thing you say about esi data about industry doesn't make any sense at all. This is just not how seat-alliance-industry works.

**Comment by martinskalicky** [comment]
From the whole topic I would be interested into:

""Regarding monthly assignments, I guess adding a feature that creates a new order on a monthly basis is enough?""
""Statistics are already requested here: https://github.com/recursivetree/seat-alliance-industry/issues/11 . maybe discuss it there?""

The rest look at least to me out of scope of this plugin.

**Comment by recursivetree** [comment]
If multiple people are interested in monthly assignments, I'll probably implement that next or at least soonish.

Statistics are a rather larger feature that in my opinion doesn't add much. You might be able to convince me if you can come up with what statistics and how they  are useful.

**Comment by dstevens79** [comment]
Accountability for who did what on a Corp level
